src/DataWrangling/ECCO/ECCO_metadata.jl

```diff
@@ -37,9 +37,24 @@ function ECCOMetadatum(name;
     return Metadatum(name; date, dir, dataset=ECCO4Monthly())
 end
 
-default_download_directory(::Union{<:ECCO2Monthly}) = joinpath(download_ECCO_cache, "v2", "monthly")
-default_download_directory(::Union{<:ECCO2Daily})   = joinpath(download_ECCO_cache, "v2", "daily")
-default_download_directory(::Union{<:ECCO4Monthly}) = joinpath(download_ECCO_cache, "v4")
+function default_download_directory(::Union{<:ECCO2Monthly})
+    path = joinpath(download_ECCO_cache, "v2", "monthly")
+    !isdir(path) && mkpath(dir)
```

Comment:
```suggestion
    mkpath(dir)
```

the docstring says "Unlike mkdir, mkpath does not error if path
  (or parts of it) already exists."